### PR TITLE
Feature: distance filter

### PR DESCRIPTION
**Pull request by bjornmosten**

### CODE REVIEW

File: server/src/util.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+///Returns a bool representing whether a vehicle is close enough to be sent out.
+///To be used with a filter function.
+pub fn filter_position(client_geo: &GeoPosition, vhc: &Vehicle) -> bool {
+    let v_pos = Location::new(vhc.position.latitude, vhc.position.longitude);
+    let client_pos = Location::new(
+        client_geo.position.coordinates[0],
+        client_geo.position.coordinates[1],
+    );
+    let distance = v_pos.distance_to(&client_pos).unwrap();
+    //(v_pos.latitude, v_pos.longitude;
```
Comment (joelsiks) [inline server/src/util.rs]: Should probably be removed

File: server/src/util.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//! Utilities. Functions that do not belong anywhere else.
+use crate::protocol::client_protocol::GeoPosition;
+use crate::protocol::server_protocol::{ServerOutput, Vehicle, VehiclePositionsOutput};
+use geoutils::Location;
+
+///Returns a bool representing whether a vehicle is close enough to be sent out.
+///To be used with a filter function.
+pub fn filter_position(client_geo: &GeoPosition, vhc: &Vehicle) -> bool {
```
Comment (joelsiks) [inline server/src/util.rs]: Would be good to rename this to `filter_vehicle_positions` or something similar, to make it clear what type of positions we are filtering.

File: server/src/lobby.rs [hunk 3]
```diff
@@ -180,6 +173,32 @@ impl Actor for Lobby {
...
+                    self.send_message(
+                        &serde_json::to_string(&ServerOutput::VehiclePositions(
+                            VehiclePositionsOutput {
+                                timestamp: Lobby::get_current_timestamp(),
+                                vehicles: filtered_vhcs,
+                            },
+                        ))
+                        .unwrap(),
+                        client_id,
+                    );
```
Comment (joelsiks) [inline server/src/lobby.rs]: Not entirely sure, but if the `filtered_vhcs` vector is empty after the filter a message is sent regardless if the client can actually "see" anything on their map. We should add a check to see if the `filtered_vhcs` is empty or not and only send data if it contains something.

**Review by Caw3 — APPROVED** [review]

**Review by joelsiks — APPROVED** [review]
👍

**Pull request merged**